models/DataObject/Classificationstore.php

```diff
@@ -180,6 +189,26 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
 
         $keyConfig = Model\DataObject\Classificationstore\DefinitionCache::get($keyId);
         $dataDefinition = Model\DataObject\Classificationstore\Service::getFieldDefinitionFromKeyConfig($keyConfig);
+
+
+        if (!$this->isFieldDirty('_self')) {
+            if ($this->object) {
+                $oldData = $this->items[$groupId][$keyId][$language];
+                $oldData = $dataDefinition->getDataForResource($oldData, $this->object);
```

Comment:
is this really a performance optimization if it queries the database for two values?

Reply:
please clarify ... getDataForResource vs getDataFromResource ... line 197

Reply:
what I mean is: you are hitting the database twice here to check if something has changed.

Reply:
actually getDataForResource doesn't hit the database at all. it just takes the "old data" and transforms it into a format that can be written into the database (which isn't the case here)

Reply:
alright then :)